src/common/services/api/order.service.js

```diff
@@ -399,6 +402,55 @@ class Order {
       return startedOrderWithoutSpouse
     }
   }
+
+  submitOrder (controller) {
+    delete controller.submissionError
+    delete controller.submissionErrorStatus
+    // Prevent multiple submissions
+    if (controller.submittingOrder) {
+      return Observable.empty()
+    }
+    controller.submittingOrder = true
+    controller.onSubmittingOrder({ value: true })
+
+    let submitRequest
+    if (controller.bankAccountPaymentDetails) {
+      submitRequest = this.submit()
+    } else if (controller.creditCardPaymentDetails) {
+      const cvv = this.retrieveCardSecurityCode()
+      submitRequest = this.submit(cvv)
+    } else {
+      submitRequest = Observable.throw({ data: 'Current payment type is unknown' })
+    }
+    return submitRequest
+      .do(() => {
```

Comment:
You should add imports for the `do` and `catch` operators as well.

Reply:
Actually, `do` accepts a second argument, which is the error handler. I think using it instead of `.catch` would be better since you can add a success and error handler at the same time.